orbit/pkg/update/flag_runner.go

```diff
@@ -126,6 +126,191 @@ func (r *FlagRunner) DoFlagsUpdate() (bool, error) {
+
+	for {
+		select {
+		case <-r.cancel:
+			return nil
+		case <-ticker.C:
+			log.Debug().Msg("calling /config API to fetch/update extensions")
+			didExtensionsUpdate, err := r.DoExtensionConfigUpdate()
+			if err != nil {
+				log.Info().Err(err).Msg("ext update failed")
+			}
+			if didExtensionsUpdate {
+				log.Info().Msg("successfully updated/fetched extensions from /config API")
+				return nil
+			}
+		}
+		ticker.Reset(r.opt.CheckInterval)
+	}
+}
+
+// Interrupt is the oklog/run interrupt method that stops orbit when interrupt is received
+func (r *ExtensionRunner) Interrupt(err error) {
+	close(r.cancel)
+	log.Debug().Err(err).Msg(("interrupt extension runner"))
+}
+
+// DoExtensionConfigUpdate calls the /config API endpoint to grab extensions from Fleet
+// It parses the extensions, computes the local hash, and writes the binary path to extension.load file
+//
+// It returns a (bool, error), where bool indicates whether orbit should restart
+func (r *ExtensionRunner) DoExtensionConfigUpdate() (bool, error) {
+	// call "/config" API endpoint to grab orbit configs from Fleet
+	config, err := r.configFetcher.GetConfig()
+	if err != nil {
+		// we do not want orbit to restart
+		return false, fmt.Errorf("extensionsUpdate: error getting extensions config from fleet: %w", err)
+	}
+
+	extensionAutoLoadFile := filepath.Join(r.opt.RootDir, "extensions.load")
+	if len(config.Extensions) == 0 {
+		// Extensions from Fleet is empty
+		// this can be either because of:
+		// 1. the default state, where no extensions are configured to begin with, or
+		// 2. extensions were previously configured, but now are deleted and reverted to empty state
+		switch stat, err := os.Stat(extensionAutoLoadFile); {
+		// Handle case 1, where our autoload file does not exist, so there is nothing to update and no error
+		case errors.Is(err, os.ErrNotExist):
+			log.Debug().Msg(extensionAutoLoadFile + " not found, nothing to update")
+			// we do not want orbit to restart
+			return false, nil
+		case err == nil:
+			// handle case 2: create/truncate the extensions.load file and let the runner interrupt, so that
+			// osquery can't startup without the extensions that were previously loaded
+			// WriteFile will create the file if it doesn't exist, and it handles Close for us
+			if stat.Size() > 0 {
+				err := os.WriteFile(extensionAutoLoadFile, []byte(""), constant.DefaultFileMode)
+				if err != nil {
+					// we do not want orbit to restart
+					return false, fmt.Errorf("extensionsUpdate: error creating file %s, %w", extensionAutoLoadFile, err)
+				}
+				// we want to return true here, and restart with the empty extensions.load file
+				// so that we "unload" the previously loaded extensions
+				return true, nil
+			}
+			// we do not want orbit to restart
+			return false, nil
+		default:
+			// we do not want orbit to restart, just log the error
+			log.Debug().Err(err).Msg("extensionsUpdate: error stating file at " + extensionAutoLoadFile)
```

Comment:
This is logging the error on `debug` mode only.

Let's return `return false, fmt.Errorf("stat file: %s", extensionAutoLoadFile)`

The caller `ExtensionRunner.Execute` will log the error.